torch/fx/experimental/proxy_tensor.py

```diff
@@ -152,10 +163,11 @@ def wrap_key(f, inps):
     @functools.wraps(f)
     def wrapped(*args):
         flat_args, args_spec = pytree.tree_flatten(args)
-        assert(len(flat_args) == len(flat_inps))
+        assert (len(flat_args) == len(flat_inps))
         for idx, arg in enumerate(flat_args):
             if isinstance(flat_inps[idx], torch.Tensor):
-                flat_args[idx] = ProxyTensor(flat_inps[idx], arg)
+                with no_dispatch():
```

Comment:
weird

Reply:
Why is this needed?

Reply:
yup

Reply:
Parroting @zou3519 from offline (though he can probably explain better):

This might not work if the vanilla tensor's dict gets cleared from other operations or interactions

We then discussed the idea of having a mapping from tensor to proxy in the mode, which could also eliminate ProxyTensor. I worried about this because right now if we have two instances of a proxy mode with the same tracer, it would continue the trace (and this behavior seems like something that might help AOTAutograd with composable behavior in the future (i.e. `aot_autograd(aot_autograd`)?)